lib/trino-orc/src/main/java/io/trino/orc/OrcWriteValidation.java

```diff
@@ -628,6 +630,11 @@ else if (VARBINARY.equals(type) || UUID.equals(type)) {
                 fieldExtractor = ignored -> ImmutableList.of();
                 fieldBuilders = ImmutableList.of();
             }
+            else if (TIME_MICROS.equals(type)) {
+                statisticsBuilder = new TimeStatisticsBuilder(new NoOpBloomFilterBuilder());
```

Comment:
I think we're supposed to use the same class for building statistics in the writer and in the validation. Since we're not changing the writer, I think this should use IntegerStatisticsBuilder. Would dropping the `/ PICOSECONDS_PER_MICROSECOND` in other places in the tests allow using `IntegerStatisticsBuilder` here ?

Reply:
@raunaqmorarka Using `IntegerStatisticsBuilder` leads to overflow when calculating sum as far as I confirmed. Could you look into the details?

Reply:
I added a change to use TimeMicrosStatisticsBuilder in ColumnWriters instead. This makes the writer consistent with the validation logic